app/services/ecr_service.rb

```diff
@@ -0,0 +1,49 @@
+class EcrService
+  def validate_image!(heritage)
+    @heritage = heritage
+
+    # The string after the last / will be matched.
+    # For example, when the Image name is public.ecr.aws/degica/barcelona,
+    # it will return barcelona.
+    repository_name = @heritage.image_name[%r{/([^/]*?)$}, 1]
```

Comment:
Thanks for the explanation. This comment is a test case so you can actually make this a method and test it.

```
  def repository_name
    @heritage.image_name[%r{/([^/]*?)$}, 1]
```

```
  ecrservice.new(Heritage.new(image_name:'public.ecr.aws/degica/barcelona'))
  expect(ecrservice.repository_name).to eq 'barcelona'
```

Also `*` is a zero-or-more matcher so there is no need to do a non-greedy `*?`. Just go `+` will be fine. `%r{/([^/]+)$}`

Reply:
thanks!
updated it https://github.com/degica/barcelona/pull/703/commits/3cd0291f8e4148297ed33033d1835bf438bea9ff